### PR TITLE
Save your nick and key aliases to a config file

### PR DESCRIPTION
**Pull request by voidcase**
Added options for setting nick and making aliases for keys.

    cabal --config-nick voidcase

    cabal --new-alias meta --key dat://21b2b9ff201b01e6081709d82e6b81a5cf3a68d2cd5f092d0ffec58772642892
Aliases are used with

     cabal meta

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -17,26 +19,73 @@ var usage = `Usage
...
+    process.stderr.write(usage)
+    process.exit(1)
+  }
+}
+
+if (args['config-nick']) {
+  config.nick = args['config-nick']
+  saveConfig(config)
+  process.stdout.write("nick " + args['config-nick'] + " saved.\n")
+  process.exit(0)
```
Comment (soyuka) [inline cli.js]: can't we continue here (open cabal)?

Comment (voidcase) [reply cli.js]: Sure, just gotta make sure the user doesn't get the help text if there was no --key or --db supplied when they use --config-nick

**Comment by voidcase** [comment]
There, changed some names also. Thought it would be nice to begin all config options with "set-"

File: cli.js [hunk 2]
```diff
@@ -17,26 +19,74 @@ var usage = `Usage
...
+
+  cabal youraliashere
+
   Options:
 
-    --nick    Your nickname.
-    --seeder  Start a headless seeder for the specified cabal key
+    --nick <name>         Use <name> as nick for this session
+    --seeder              Start a headless seeder for the specified cabal key
+    --set-alias <name>    Add an alias to your config. Must be used with --key
```
Comment (Powersource) [inline cli.js]: Could clarify that it's an alias for a key.

**Comment by karissa** [comment]
can we use something like git, where it is `config set <key> <value>` ? This will be more future proof if we want to add keys later, and easier to guess! 
https://www.git-scm.com/book/en/v2/Customizing-Git-Git-Configuration

**Comment by voidcase** [comment]
@karissa 
I agree! But it would be weird for "config" to occupy the same namespace as the users aliases. so we should probably change some of the current options to subcommands for consistency.
For example:

`
cabal open dat://21b2b9ff201b01e6081709d82e6b81a5cf3a68d2cd5f092d0ffec58772642892
cabal set keyalias.meta 21b2b9ff201b01e6081709d82e6b81a5cf3a68d2cd5f092d0ffec58772642892
cabal open meta
`
What do you think the CLI should look like, @cblgh ? Since this is a pretty big change. It seems like using options for everything might become too clumsy pretty fast.

**Comment by noffle** [comment]
I'm in favour of this kind of functionality being built into a common `cabal-client` module, since it's such a useful piece of functionality that every client will want to implement it.

**Comment by cblgh** [comment]
+1

**Comment by cblgh** [comment]
closing this as we should implement it in [cabal-client](https://github.com/cabal-club/cabal)

**Pull request closed without merging**